configure.ac

```diff
@@ -719,6 +719,11 @@ if test "$with_skey" = "yes"; then
 	]])],[AC_DEFINE(SKEY_BSD_STYLE, 1, [Define to support newer BSD S/Key API])],[])
 fi
 
+AC_SUBST(CMOCKA_LIBS)
+AC_CHECK_LIB(cmocka, _will_return, [CMOCKA_LIBS=-lcmocka]; [have_cmocka="yes"],
+	[AC_MSG_WARN([libcmocka not found, cmocka tests will not be built])])
```

Comment:
This will just check whether library is available for linking
* https://www.gnu.org/software/autoconf/manual/autoconf-2.67/html_node/Libraries.html
And that would pass just with the **libcmocka0** installed on system
* https://packages.debian.org/sid/amd64/libcmocka0/filelist

One should check also availability of header files.

Or you can rely on pkgconfig which is already provided by cmocka and it makes configure time check much simpler.
* https://packages.debian.org/sid/amd64/libcmocka-dev/filelist

and it is already used in the same file
* https://github.com/shadow-maint/shadow/blob/5178f8c5afb612f6ddf5363823547e080e7f546b/configure.ac#L424

Reply:
I preferred the `PKG_CHECK_MODULES` way, so I've used it. Take a look at it to check that I got it right.

Reply:
LGTM